lib/xcodeproj/plist_helper.rb

```diff
@@ -450,3 +478,166 @@ def self.RubyBooleanToCFBoolean(value)
   # rubocop:enable Style/MethodName
   # rubocop:enable Style/VariableName
 end
+
+module DevToolsCore
+  def self.silence_stderr
+    begin
+      orig_stderr = $stderr.clone
+      $stderr.reopen File.new('/dev/null', 'w')
+      retval = yield
+    ensure
+      $stderr.reopen orig_stderr
+    end
+    retval
+  end
+
+  # rubocop:disable Style/MethodName
+  # rubocop:disable Style/VariableName
+
+  class NSObject
+    private
+
+    def self.objc_class
+      @objc_class ||= CoreFoundation.objc_getClass(name.split('::').last)
+    end
+
+    def self.image
+      @image ||= Fiddle::Handle.new
+    end
+
+    def self.extern(symbol, parameter_types, return_type)
+      CoreFoundation.extern_image(image, symbol, parameter_types, return_type)
+    end
+
+    def self.objc_msgSend(args, return_type = CoreFoundation::VoidPointer)
+      arguments = [CoreFoundation::VoidPointer, CoreFoundation::VoidPointer] + args
+
+      Fiddle::Function.new(image['objc_msgSend'], arguments, return_type)
+    end
+
+    def self.respondsToSelector(instance, sel)
+      selector = CoreFoundation.NSSelectorFromString(CoreFoundation.RubyStringToCFString(sel))
+      respondsToSelector = objc_msgSend([CoreFoundation::CharPointer], CoreFoundation::Boolean)
+      result = respondsToSelector.call(
+        instance,
+        CoreFoundation.NSSelectorFromString(CoreFoundation.RubyStringToCFString('respondsToSelector:')),
+        selector)
+      result == CoreFoundation::TRUE ? true : false
+    end
+
+    Class = CoreFoundation::VoidPointer
+    ID = CoreFoundation::VoidPointer
+    SEL = CoreFoundation::VoidPointer
+
+    extern :NSSelectorFromString, [CoreFoundation::CFTypeRef], SEL
+
+    extern :objc_getClass, [CoreFoundation::CharPointer], Class
+    extern :class_getName, [Class], CoreFoundation::CharPointer
+  end
+
+  XCODE_PATH = Pathname.new(`xcrun xcode-select -p`.strip).dirname
```

Comment:
I'll add a spec for this. IMHO, we should silently fallback to XML writing, just as in other cases of not being able to use the private Xcode API.